toolkit/tools/pkg/imagecustomizerlib/customizepackages_test.go

```diff
@@ -211,3 +214,120 @@ func TestCustomizeImagePackagesDiskSpace(t *testing.T) {
 	assert.ErrorContains(t, err, "failed to customize raw image")
 	assert.ErrorContains(t, err, "failed to install package (gcc)")
 }
+
+func ensureTdnfCacheCleanup(t *testing.T, imageConnection *ImageConnection, dirPath string) error {
+	// Get version of base image
+	osVersion := getImageVersion(imageConnection)
+
+	// Variables to capture root directory contents
+	var rootDirContents []string
+	var foundLocalRepo, foundOfficialBase bool
+
+	entries, err := os.ReadDir(filepath.Join(imageConnection.Chroot().RootDir(), dirPath))
+	if err != nil {
+		return fmt.Errorf("Failed to read directory (%s): %v", dirPath, err)
+	}
+
+	for _, entry := range entries {
+		if strings.Contains(entry.Name(), "local-repo") {
+			foundLocalRepo = true
+		}
+		if strings.HasSuffix(entry.Name(), "official-base") {
+			foundOfficialBase = true
+		}
+		entryFullPath := filepath.Join(dirPath, entry.Name())
+		rootDirContents = append(rootDirContents, entryFullPath)
+	}
+
+	// Ensure there are only the expected number of folders after the tdnf cleanup
+	if strings.HasPrefix(osVersion, "2.0") {
+		assert.True(t, foundLocalRepo, "Expected to find a folder containing 'local-repo'")
+		assert.True(t, foundOfficialBase, "Expected to find a folder ending with 'official-base'")
+		assert.Equal(t, 2, len(rootDirContents), "Expected exactly 2 folders, but got %d", len(rootDirContents))
+	}
+	if strings.HasPrefix(osVersion, "3.0") {
+		assert.True(t, foundOfficialBase, "Expected to find a folder ending with 'official-base'")
+		assert.Equal(t, 1, len(rootDirContents), "Expected exactly 1 folders, but got %d", len(rootDirContents))
+	}
+
+	// Search the cache directory and all subdirectories and extract all contents
+	dirContents, err := searchChrootDirectory(imageConnection, dirPath)
+	if err != nil {
+		return err
+	}
+
+	// Filter and collect non-directory files, excluding those related to 'local-repo'
+	var existingFiles []string
+	for _, entry := range dirContents {
+		if !strings.Contains(entry, "local-repo") {
+			fileInfo, err := os.Stat(entry)
```

Comment:
If you changed `searchChrootDirectory` to only return files, then you wouldn't need to do this extra step here.